org.eclipse.jdt.ls.core/src/org/eclipse/jdt/ls/core/internal/contentassist/CompletionProposalRequestor.java

```diff
@@ -323,6 +324,11 @@ private void initializeCompletionListItemDefaults(CompletionProposal proposal) {
 		if (completionItem.getTextEdit() != null && preferenceManager.getClientPreferences().isCompletionListItemDefaultsEditRangeSupport()) {
 			itemDefaults.setEditRange(getEditRange(completionItem, preferenceManager));
 		}
+		if (preferenceManager.getClientPreferences().isCompletionListItemDefaultsInsertTextModeSupport() &&
+			preferenceManager.getClientPreferences().getCompletionItemInsertTextModeDefault() != InsertTextMode.AdjustIndentation
+		) {
+			itemDefaults.setInsertTextMode(InsertTextMode.AdjustIndentation);
```

Comment:
What if the client does not support `AdjustIndentation`?

In other words: `AdjustIndentation` does not declared in `insertTextModeSupport` array.

Reply:
I would think if the client doesn't support the option that our LS strongly prefers (`AdjustIndentation`), then we can let the client do as it wishes (situation we have today).

Reply:
I''ve added an additional check here